### PR TITLE
UIDEXP-51: Implement SearchForm component

### PR DESCRIPTION
**Pull request by VictorSoroka1**
## Purpose

Add `SearchForm` component in the scope of [UIDEXP-51](https://issues.folio.org/browse/UIDEXP-51).

## Screenshot

Screen Shot 2020-03-26 at 16 07 42

### CODE REVIEW

File: lib/SearchForm/SearchForm.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+};
+
+SearchForm.defaultProps = {
+  isLoading: false,
+  searchableIndexes: null,
+  selectedIndex: '',
+  handleChangeIndex: noop,
+  handleChange: noop,
+  handleSubmit: noop,
+  handleClear: null,
```
Comment (YevheniiMaltsev) [inline lib/SearchForm/SearchForm.js]: Could you please clarify why noop is used for handleChangeIndex, handleChange and handleSubmit, but in this case null is used?

Comment (VictorSoroka1) [reply lib/SearchForm/SearchForm.js]: Sure, there is a check in the `SearchField` component.
```
hasClearIcon={typeof onClear === 'function' && loading !== true}
```
I think it makes sense to let the end-user decide whether to show it for consistency. I am used to always pass `noop` as default prop for functions but in this case, I think it does not make sense to do so.

Comment (YevheniiMaltsev) [reply lib/SearchForm/SearchForm.js]: Thanks. It's clear now

Comment (VictorSoroka1) [reply lib/SearchForm/SearchForm.js]: Good catch, I was sure that you would notice such kind of stuff :slightly_smiling_face:

**Pull request merged**